### PR TITLE
Scoped conductor logs

### PR DESCRIPTION
**Pull request by maackle**
### Summary

Instrument all traces from all tasks that get spawned by a conductor, according to a tag in the ConductorConfig. This is so that when we have multiple Conductors running in the same process, we can tell their logs apart.

It would be nice if we could just use the `ConductorStateTag` in the state, but this needs to be unique, and this scoping tag is meant to be human readable foremost, and also as unique as the user needs, so the constraints are slightly different.

### TODO:
- [ ] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/holochain/src/conductor/manager/mod.rs [hunk 1]
```diff
@@ -295,6 +356,13 @@ fn handle_completed_task(kind: &TaskKind, result: ManagedTaskResult, name: Strin
             Ok(_) => LogInfo(name),
             Err(err) => MinorError(Box::new(err), name),
         },
+        TaskKind::Persistent => match result {
+            Ok(_) => LogInfo(name),
+            Err(err) => {
+                tracing::warn!("Restarting persistent tasks is not yet implemented.");
+                MinorError(Box::new(err), name)
```
Comment (neonphog) [inline crates/holochain/src/conductor/manager/mod.rs:363]: Should this be fatal for now, if we're depending on the task being persistent but we don't have the restart mechanism yet?

File: crates/holochain/src/conductor/manager/mod.rs [hunk 2]
```diff
@@ -145,13 +121,99 @@ impl TaskManager {
...
+
+    /// You just want the task in the task manager but don't want
+    /// to react to an error
+    pub async fn ignored_task(&self, name: &str, fut: impl ManagedTaskFut) -> TaskAddResult {
+        self.add_task(name, TaskKind::Ignore, fut).await
+    }
+
+    /// You just want the task in the task manager but don't want
+    /// to react to an error
+    pub async fn persistent_task(&self, name: &str, fut: impl ManagedTaskFut) -> TaskAddResult {
```
Comment (neonphog) [inline crates/holochain/src/conductor/manager/mod.rs:170]: maybe update the comments on this one--looks like it's a copy/paste from ignored_task

File: crates/holochain/src/conductor/manager/mod.rs [hunk 3]
```diff
@@ -145,13 +121,99 @@ impl TaskManager {
...
+    /// Add a task of a certain kind
+    pub async fn add_task(
+        &self,
+        name: &str,
+        kind: TaskKind,
+        fut: impl ManagedTaskFut,
+    ) -> TaskAddResult {
+        let span = tracing::error_span!("conductor", scope = self.tag);
+        let handle = tokio::spawn(fut.instrument(span));
+        self.tx.send(ManagedTask::new(name, kind, handle)).await
```
Comment (neonphog) [inline crates/holochain/src/conductor/manager/mod.rs:159]: :+1: - I'm still a little bummed the scope isn't inherited when spawning... And yeah, still wondering if it'd be worth it to introduce a runtime abstraction to both kitsune and conductor so we could work around this...

**Review by neonphog — APPROVED** [review]
:+1:

**Comment by maackle** [comment]
Unfortunately this PR is now woefully out of date. It's probably best to start over.

The salvageable idea from this PR is that the task manager needs to use `.instrument(span)` on every task which it manages, so that the logs contains the proper context.

see:
- adding a Config item for the "scope" of this conductor to be used in logs https://github.com/holochain/holochain/pull/1623/files#diff-7e6690c5291c8fe44ddac29258eabff9db1daf2cd5e83883dd48b933b84d69c4R68
- instrumenting managed tasks: https://github.com/holochain/holochain/pull/1623/files#diff-089f64a9ad80beecf5d9199c5745d9477affc02918662436d1f7851b9d0a1be6R158

**Comment by jost-s** [comment]
Salvaged and superseded by #2881 
@maackle can we close this?

**Pull request closed without merging**